### PR TITLE
Add buffers argument to comm.open/close

### LINKED ISSUE #2556: Make opening a comm accept buffers
Right now when we open a comm, we can set the data and metadata of the message, but we can't send buffers. It should be straightforward to add the buffers optional argument to the open call at https://github.com/jupyterlab/jupyterlab/blob/080a21f5a4d8e7a90f5b15572baa4add7776c3d1/packages/services/src/kernel/comm.ts#L105

### PR DESCRIPTION
**Pull request by vidartf**
Adds a `buffers` argument to services.kernel.Comm.open/close.

I'm a little unsure about the `ioMsg` create for `onClose` in the close method, but other than that it should be straight forward since it uses the same internal methods as the `send` method.

Fixes #2556.

### CODE REVIEW

File: packages/services/src/kernel/comm.ts [hunk 1]
```diff
@@ -165,12 +165,12 @@ class CommHandler extends DisposableDelegate implements Kernel.IComm {
...
       data: data || {}
     };
-    let msg = KernelMessage.createShellMessage(options, content, metadata);
+    let msg = KernelMessage.createShellMessage(options, content, metadata, buffers);
     let future = this._kernel.sendShellMessage(msg, false, true);
     options.channel = 'iopub';
-    let ioMsg = KernelMessage.createMessage(options, content, metadata);
     let onClose = this._onClose;
     if (onClose) {
+      let ioMsg = KernelMessage.createMessage(options, content, metadata, buffers);
```
Comment (vidartf) [inline packages/services/src/kernel/comm.ts]: This is the message whose purpose I'm a little unsure about.

**Comment by vidartf** [comment]
Update: I figured out the question I had about the original test, and changed it back.

**Comment by blink1073** [comment]
Terminal test looks related to a recent release of xterm, looking into that now.

**Comment by blink1073** [comment]
@vidartf, mind rebasing to pick up https://github.com/jupyterlab/jupyterlab/pull/2793?

**Comment by vidartf** [comment]
Rebased, tests running.

**Comment by blink1073** [comment]
Thanks!

**Pull request merged**